flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/restore/RocksDBIncrementalRestoreOperation.java

```diff
@@ -539,13 +552,14 @@ private void rescaleClipIngestDB(
             }
 
             // If there are temporary instances for which we detected overlap in the buffer or
-            // if we have remaining handles to restore, we will insert by copy with from temporary
-            // instances to base DB.
+            // if we have remaining unopened handles to restore, we will insert by copy via
+            // temporary instances to base DB.
```

Comment:
In the old code, we called `choseTheBestStateHandleForInitial` to choose the best state handle to init the initial db .Because if we use TheBestHandle as a temporary DB instead of the main DB, we may need to write a lot of data when copying. Can we maintain this logic int the new code ?
When choosing db to export, we prioritize `TheBestHandle` to ensure that we will not copy TheBestHandle during the subsequent copying phase. Can this ensure that there is no regression compared to the old code

Reply:
Yes, I'm already working on adding that code again here.